src/main/java/com/truelayer/java/http/mappers/HeadersMapper.java

```diff
@@ -0,0 +1,40 @@
+package com.truelayer.java.http.mappers;
+
+import static org.apache.commons.lang3.ObjectUtils.isEmpty;
+import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
+
+import com.truelayer.java.Constants;
+import com.truelayer.java.http.entities.Headers;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Utility class for Headers to Map conversions.
+ */
+public class HeadersMapper {
```

Comment:
I would have loved to use the default visibility here but I would need to restructure a lot.